carma_cooperative_perception/src/msg_conversion.cpp

```diff
@@ -265,4 +382,150 @@ auto to_external_object_list_msg(
+        detected_object_optional_data.det_veh.presence_vector |= carma_v2x_msgs::msg::DetectedVehicleData::HAS_HEIGHT;
+
+        detected_object_optional_data.det_veh.size.vehicle_width = external_object.size.y;
+        detected_object_optional_data.det_veh.size.vehicle_length = external_object.size.x;
+        detected_object_optional_data.det_veh.height.vehicle_height = external_object.size.z;
+      }
+      break;
+    case external_object.LARGE_VEHICLE:
+      detected_object_common_data.obj_type.object_type = j3224_v2x_msgs::msg::ObjectType::VEHICLE;
+
+      if(external_object.presence_vector & external_object.SIZE_PRESENCE_VECTOR){
+        detected_object_optional_data.det_veh.presence_vector |= carma_v2x_msgs::msg::DetectedVehicleData::HAS_SIZE;
+        detected_object_optional_data.det_veh.presence_vector |= carma_v2x_msgs::msg::DetectedVehicleData::HAS_HEIGHT;
+
+        detected_object_optional_data.det_veh.size.vehicle_width = external_object.size.y;
+        detected_object_optional_data.det_veh.size.vehicle_length = external_object.size.x;
+        detected_object_optional_data.det_veh.height.vehicle_height = external_object.size.z;
+      }
+      break;
+    case external_object.MOTORCYCLE:
+      detected_object_common_data.obj_type.object_type = j3224_v2x_msgs::msg::ObjectType::VEHICLE;
+
+      if(external_object.presence_vector & external_object.SIZE_PRESENCE_VECTOR){
+        detected_object_optional_data.det_veh.presence_vector |= carma_v2x_msgs::msg::DetectedVehicleData::HAS_SIZE;
+        detected_object_optional_data.det_veh.presence_vector |= carma_v2x_msgs::msg::DetectedVehicleData::HAS_HEIGHT;
+
+        detected_object_optional_data.det_veh.size.vehicle_width = external_object.size.y;
+        detected_object_optional_data.det_veh.size.vehicle_length = external_object.size.x;
+        detected_object_optional_data.det_veh.height.vehicle_height = external_object.size.z;
+      }
+      break;
+    case external_object.PEDESTRIAN:
+      detected_object_common_data.obj_type.object_type = j3224_v2x_msgs::msg::ObjectType::VRU;
+
+      detected_object_optional_data.det_vru.presence_vector |= carma_v2x_msgs::msg::DetectedVRUData::HAS_BASIC_TYPE;
+      detected_object_optional_data.det_vru.basic_type.type |= j2735_v2x_msgs::msg::PersonalDeviceUserType::A_PEDESTRIAN;
+
+      break;
+    case external_object.UNKNOWN:
+    default:
+      detected_object_common_data.obj_type.object_type = j3224_v2x_msgs::msg::ObjectType::UNKNOWN;
+
+      if(external_object.presence_vector & external_object.SIZE_PRESENCE_VECTOR){
+
+        detected_object_optional_data.det_obst.obst_size.width.size_value = external_object.size.y;
+        detected_object_optional_data.det_obst.obst_size.length.size_value = external_object.size.x;
+
+        detected_object_optional_data.det_obst.obst_size.presence_vector |= carma_v2x_msgs::msg::ObstacleSize::HAS_HEIGHT;
```

Comment:
The `presence_vector` should be zeroed out initially to avoid undefined initial value issues.